contracts/access/manager/IAuthority.sol

```diff
@@ -0,0 +1,7 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.20;
+
+interface IAuthority {
+    function canCall(address caller, address target, bytes4 selector) external view returns (bool allowed, uint32 delay);
```

Comment:
When the governor plans a call on some contract, and the contract has a manager, we are going to call `canCall` on that manager to check the permissions. This will be decoded as `(bool, uint32)` because we need the delay.

Are you suggesting that we do the decoding manually depending on the return length ? and that if the return length is more than a give value we assume what is beyond the interface's `bool` is a `uint32`. I find that quite dangerous.

Reply:
Yes, that's correct. What do you think is dangerous about it? 

> if the return length is more than a give value we assume what is beyond the interface's `bool` is a `uint32`

This assumption is basically the same that would be happening if we define the interface the way it's defined now, except that it reverts if the return length is less.

Is there something else?

Reply:
implemented